nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -688,3 +688,67 @@ export async function findUsableCommitmentsMutex(
     findUsableCommitments(compressedZkpPublicKey, ercAddress, tokenId, _value, onlyOne),
   );
 }
+
+/**
+ * @function getCommitmentsByCompressedPkdList do the role of a service taking care of the
+ * business logic and of a repository doing the communication with the database for this
+ * use case.
+ * @param {string[]} compressedZkpPublicKey a list of compressedZkpPublicKey derivated from the user
+ * mnemonic coming from the SDK or Wallet.
+ * @returns all the commitments existent for this list of compressedZkpPublicKey.
+ * @author luizoamorim
+ */
+export async function getAllCommitmentsByCompressedZkpPublicKeyList(listOfCompressedZkpPublicKey) {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+  const allCommitmentsByListOfCompressedZkpPublicKey = await db
+    .collection(COMMITMENTS_COLLECTION)
+    .find({
+      'preimage.compressedZkpPublicKey': { $in: listOfCompressedZkpPublicKey },
+    })
+    .toArray();
+  return allCommitmentsByListOfCompressedZkpPublicKey;
+}
+
+/**
+ * @function getAllCommitments do the role of a service taking care of the
+ * business logic and of a repository doing the communication with the database for this
+ * use case.
+ * @returns all the commitments existent in this database.
+ * @author luizoamorim
+ */
+export async function getAllCommitments() {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+  const allCommitments = await db.collection(COMMITMENTS_COLLECTION).find({}).toArray();
+  return allCommitments;
+}
+
+/**
+ *
+ * @function saveAllCommitments save a list of commitments in the database
+ * @param {[]} listOfCommitments a list of commitments to be saved in the database
+ * @returns
+ */
+export async function saveAllCommitments(listOfCommitments) {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+  let exist = false;
+  for (const commitment of listOfCommitments) {
+    if (
+      // eslint-disable-next-line no-await-in-loop
+      (await db.collection(COMMITMENTS_COLLECTION).find({ _id: commitment._id }).toArray()).length >
```

Comment:
I don't know what would be the impact of removing this verification, the `for` iteration. A `save` operation normally is meant to insert or update, then having a pre-existing object in the DB wouldn't be a problem.

Reply:
I created this thinking in the follow case:
- The user send a list of commitments: 50% already exist in the database and the other 50% doesn't exist.
- The idea was not insert nothing if have some wrong commit in the list.

But yes, if we remove. The database will generate an exception showing that the _id already exists. I just don't know if will generate the exception and will insert the commitments that don't exists. Or will generate the exception and won't insert nothing.

Reply:
So. I'll filter only the commitments that don't exist in the database yet and will insert all the others like we talked in the chat.